pkg/options/options.go

```diff
@@ -82,6 +86,13 @@ func (o *Options) AddFlags() {
 	o.flags.StringVar(&o.TelemetryHost, "telemetry-host", "0.0.0.0", `Host to expose kube-state-metrics self metrics on.`)
 	o.flags.Var(&o.Collectors, "collectors", fmt.Sprintf("Comma-separated list of collectors to be enabled. Defaults to %q", &DefaultCollectors))
 	o.flags.Var(&o.Namespaces, "namespace", fmt.Sprintf("Comma-separated list of namespaces to be enabled. Defaults to %q", &DefaultNamespaces))
+	o.flags.Uint64Var(&o.Shard, "shard", uint64(0), "The instances shard nominal (zero indexed) within the total number of shards. (default 0)")
+	o.flags.Uint64Var(&o.TotalShards, "total-shards", uint64(1), "The total number of shards. Sharding is disabled when total shards is set to 1.")
+
+	autoshardingNotice := "When set, it is expected that --pod and --pod-namespace are both set. This is used for auto-detecting sharding. If set this has preference over statically configured sharding. This is exprimental, it may be removed without notice."
```

Comment:
```suggestion
	autoshardingNotice := "When set, it is expected that --pod and --pod-namespace are both set. This is used for auto-detecting sharding. If set, this has preference over statically configured sharding. This is experimental, it may be removed without notice."
```